### PR TITLE
SFI-2270 Implement refined statement constructor database design

### PR DESCRIPTION
**Pull request by AbidemiAdio**

### CODE REVIEW

File: app/data/models/funding-option.js [hunk 1]
```diff
@@ -1,8 +1,7 @@
 module.exports = (sequelize, DataTypes) => {
   const fundingOption = sequelize.define('fundingOption', {
     fundingCode: { type: DataTypes.STRING, primaryKey: true, autoIncrement: true },
```
Comment (marctemp) [inline app/data/models/funding-option.js:3]: what  mad man made this auto-incr?!

`fundingCode` is like `80001` being Arable soil: Intro

could you remove please?

File: app/data/models/funding.js [hunk 2]
```diff
@@ -3,7 +3,8 @@ module.exports = (sequelize, DataTypes) => {
     fundingId: { type: DataTypes.INTEGER, primaryKey: true },
```
Comment (marctemp) [inline app/data/models/funding.js:3]: Again, another mistake not on yer part -- this should have auto-incr prop

File: changelog/db.changelog-1.5.xml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:pro="http://www.liquibase.org/xml/ns/pro" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/pro http://www.liquibase.org/xml/ns/pro/liquibase-pro-3.9.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.9.xsd">
+ <changeSet author="Abidemi Adio" id="1">
+    <tagDatabase tag="v1.4.0" />
+  </changeSet>
+  <changeSet author="Abidemi Adio" id="2">
+    <addColumn tableName="calculations">
+      <column name="invoiceNumber" type="VARCHAR(20)" />
```
Comment (marctemp) [inline changelog/db.changelog-1.5.xml:8]: This is `varchar(30)` this is the enriched `invoiceNumber`

File: app/data/models/calculation.js [hunk 4]
```diff
@@ -3,6 +3,7 @@ module.exports = (sequelize, DataTypes) => {
     calculationId: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
     paymentRequestId: DataTypes.INTEGER,
     sbi: DataTypes.INTEGER,
+    invoiceNumber: DataTypes.STRING,
```
Comment (marctemp) [inline app/data/models/calculation.js:6]: my OCD would suggest putting this below to be alphabetical ordered :D

File: changelog/db.changelog-1.5.xml [hunk 5]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
```
Comment (marctemp) [inline changelog/db.changelog-1.5.xml:1]: jus what John mentioned tht I imagine `main` will have `1.5` innit by the time this is merged -- just need to keep on-top of it

File: test/integration/process-return-settlement.test.js [hunk 6]
```diff
@@ -83,4 +76,11 @@ describe('process return settlement', () => {
     const result = await db.settlement.findOne({ where: { invoiceNumber: settlement.invoiceNumber } })
     expect(result.sourceSystem).toBeUndefined()
   })
+
+  test('should not save frn in the table when valid settlement received ', async () => {
```
Comment (marctemp) [inline test/integration/process-return-settlement.test.js:80]: NICE!! Well remembered man

**Review by marctemp — CHANGES_REQUESTED** [review]
Couple of things missing:

- we decided to remove the `frn` link between `paymentRequests` and `organisations` as we don't care about this until Vet Visits is implemented, therefore
- remove PK-FK link
- remove `frn` from `paymentRequests`
- remove FK contraint on `frn` in `organisations`
- add test that `frn` is not saved down to `paymentRequests`

File: changelog/db.changelog-1.5.xml (hunk 3 above)
Comment (marctemp) [reply changelog/db.changelog-1.5.xml:8]: NOPE! I'm wrong. 

This is correct, could you rectify this in ERD(it says `varchar(30)`

File: changelog/db.changelog-1.5.xml (hunk 5 above)
Comment (AbidemiAdio) [reply changelog/db.changelog-1.5.xml:1]: No 1.5 in main yet

File: test/integration/process-submit-payment-request.test.js [hunk 7]
```diff
@@ -517,4 +517,15 @@ describe('process submit payment request', () => {
...
+
+  test('should not save frn in paymentRequests table when valid payment request received', async () => {
+    await processSubmitPaymentRequest(paymentRequest)
+
+    const result = await db.paymentRequest.findOne({ where: { referenceId: paymentRequest.referenceId } })
+    expect(result.invoiceNumber).not.toBeUndefined()
+    expect(result.schemeId).not.toBeUndefined()
+    expect(result.marketingYear).not.toBeUndefined()
+    expect(result.sourceSystem).not.toBeUndefined()
+    expect(result.frn).toBeUndefined()
```
Comment (marctemp) [inline test/integration/process-submit-payment-request.test.js:529]: this is the only expectation we need -- the others are checked above

**Review by marctemp — APPROVED** [review]
Good to go to the testers 👍

**Review by johnwatson484 — APPROVED** [review]

**Pull request merged**